pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -181,32 +181,38 @@ func Pod(source *operatorsv1alpha1.CatalogSource, name string, image string, saN
 							corev1.ResourceMemory: resource.MustParse("50Mi"),
 						},
 					},
-					SecurityContext: &corev1.SecurityContext{
-						ReadOnlyRootFilesystem:   pointer.Bool(false),
-						AllowPrivilegeEscalation: pointer.Bool(false),
-						Capabilities: &corev1.Capabilities{
-							Drop: []corev1.Capability{"ALL"},
-						},
-					},
 					ImagePullPolicy:          pullPolicy,
 					TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
 				},
 			},
 			NodeSelector: map[string]string{
 				"kubernetes.io/os": "linux",
 			},
-			SecurityContext: &corev1.PodSecurityContext{
-				SeccompProfile: &corev1.SeccompProfile{
-					Type: corev1.SeccompProfileTypeRuntimeDefault,
-				},
-			},
 			ServiceAccountName: saName,
 		},
 	}
-
-	if runAsUser > 0 {
-		pod.Spec.SecurityContext.RunAsUser = &runAsUser
-		pod.Spec.SecurityContext.RunAsNonRoot = pointer.Bool(true)
+	if source.Spec.RunAsRoot {
+		pod.Spec.SecurityContext = &corev1.PodSecurityContext{
+			RunAsUser:    pointer.Int64(0),
+			RunAsNonRoot: pointer.Bool(false),
```

Comment:
We can leave both empty.
What we need is to set AllowPrivilegeEscalation = true 

c/c @anik120 @bparees

Reply:
/hold

Reply:
the container doesn't actually need to run as UID 0?

Reply:
does it hurt to be explicit about it them anyway?

Reply:
also /hold cancel since there's already a WIP progress tag on the PR

Reply:
/hold cancel

Reply:
> As far as I know, we never had so far any UID set (either we did not set before uid =0) and we were not setting any USERID in the Dockerfile.

if nothing is set in the dockerfile nor in the pod/container, then the container will try to run as uid 0 as long as the user's SCC permissions allow it.  Which means catalogs are probably running as uid 0 today (on older OCP versions).

So i agree with @anik120 that explicitly setting it to uid 0 is appropriate, unless someone has tested and confirmed that it works when not run as uid 0.

Reply:
@bparees,

> if nothing is set in the dockerfile nor in the pod/container, then the container will try to run as uid 0 as long as the user's SCC permissions allow it

I understand that in this case will be any user id but it will be a member from the root group
Also,  I do not think that `RunAsUser:  0,` can work on OCP and we have an implementation to ommit the runAsUser in downstream

So, ihmo we just need to make clear that no matter the user used it must have access to the root file system and only set AllowPrivilegeEscalation = true shows enough for upstream and downstream. 

However, if you folks check that it works and prefer to keep this implementation I am ok with too

/hold cancel

Reply:
> I understand that in this case will be any user id but it will be a member from the root group

it depends on the what SCC it is admitted under.  again, i'm pretty sure that historically these pods ran as uid 0, and had to.  But someone should certainly confirm.

Reply:
Fyi we're building both our [upstream](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Dockerfile#L36) and [downstream](https://github.com/openshift/operator-framework-olm/blob/master/operator-lifecycle-manager.Dockerfile#L31) images with `runAsUser: 1001`

> we have an implementation to ommit the runAsUser in downstream 

could you point us to that implementation code? I'll be extremely surprised if I as an admin want to run my worload as root by setting runAsUser to 0 but some controller in openshift says "nope, none of that business here, so what you're admin?"

Reply:
Also, for the namespaces that these CatalogSources will run, all SCC labels have to be disabled like we discussed, and the only label that'll be there for the namespace is PSA enforce:privileged, so I don't think there's any question of "workload `runAsUser` will be modified by an openshift controller" in the namespaces we care about for this scenario.

Reply:
@anik120 

> Also, for the namespaces that these CatalogSources will run, all SCC labels have to be disabled
What do you mean by disabling SCC labels? 

> so I don't think there's any question of "workload runAsUser will be modified by an openshift controller" in the namespaces we care about for this scenario.

we spoke about it before, we cannot set values for `runAsUser ` in OCP (in downstream we leave RunAsUser empty and let the OCP assign a value.)

Reply:
So this is completely different from all of the things you've mentioned in your comment. This PR is only for old CatalogSources that cannot be run in any of the openshift-* namespaces (since they're already being managed by us and have restricted profile enforce). We're dealing with the situation here where admins encounter old catalogsources they care about, so they'll create a new namespace, edit the namespace to enforce:privileged, and create the catalogsource in that namespace with `runAsRoot:true`. As simple as that. No OCP controller acting on that namespace (since we're explicitly making sure of that)

Reply:
> Fyi we're building both our [upstream](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Dockerfile#L36) and [downstream](https://github.com/openshift/operator-framework-olm/blob/master/operator-lifecycle-manager.Dockerfile#L31) images with runAsUser: 1001

we are now.  but my understanding is legacy catalog images didn't specify this, and thus run as uid==0 by default (i.e. when the pod doesn't otherwise specify a uid)

Can someone please confirm on a 4.10 install what the pod yaml looks like for the running catalog pod?  (not just what the pod yaml that the catalog controller creates, because you also have to consider what SCC admission will do during mutation)

Ultimately that is how we want the pod to look when this new field(RunAsRoot) is set to true

Reply:
> we are now. but my understanding is legacy catalog images didn't specify this, and thus run as uid==0 by default (i.e. when the pod doesn't otherwise specify a uid)

that is a good point. And I launched a 4.10 cluster and this is how the yaml looks like: 

```oc get pod community-operators-d2btv -n openshift-marketplace -o yaml 
apiVersion: v1
kind: Pod
metadata:
  annotations:
    .
    openshift.io/scc: anyuid
    operatorframework.io/managed-by: marketplace-operator
  creationTimestamp: "2022-08-29T21:24:48Z"
  generateName: community-operators-
  labels:
    catalogsource.operators.coreos.com/update: ""
    olm.catalogSource: community-operators
    olm.pod-spec-hash: 656f95f5d4
  name: community-operators-d2btv
  namespace: openshift-marketplace
  ownerReferences:
  - apiVersion: operators.coreos.com/v1alpha1
    blockOwnerDeletion: false
    controller: false
    kind: CatalogSource
    name: community-operators
    uid: f2cf0647-138d-4f72-b1e0-172ff4acb672
  resourceVersion: "31463"
  uid: acdd15f7-c191-4cf5-8174-6196bbcd000b
spec:
  containers:
  - image: registry.redhat.io/redhat/community-operator-index:v4.10
    imagePullPolicy: Always
    livenessProbe:
      exec:
        command:
        - grpc_health_probe
        - -addr=:50051
      failureThreshold: 3
      initialDelaySeconds: 10
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 5
    name: registry-server
    ports:
    - containerPort: 50051
      name: grpc
      protocol: TCP
    readinessProbe:
      exec:
        command:
        - grpc_health_probe
        - -addr=:50051
      failureThreshold: 3
      initialDelaySeconds: 5
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 5
    resources:
      requests:
        cpu: 10m
        memory: 50Mi
    securityContext:
      capabilities:
        drop:
        - MKNOD
      readOnlyRootFilesystem: false
    startupProbe:
      exec:
        command:
        - grpc_health_probe
        - -addr=:50051
      failureThreshold: 15
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: FallbackToLogsOnError
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-2vtdl
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  imagePullSecrets:
  - name: community-operators-dockercfg-bjn7x
  nodeName: ci-ln-prs6mt2-72292-7w4cb-master-0
  nodeSelector:
    kubernetes.io/os: linux
    node-role.kubernetes.io/master: ""
  preemptionPolicy: PreemptLowerPriority
  priority: 2000000000
  priorityClassName: system-cluster-critical
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext:
    seLinuxOptions:
      level: s0:c17,c4
  serviceAccount: community-operators
  serviceAccountName: community-operators
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 120
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 120
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  volumes:
  - name: kube-api-access-2vtdl
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
      - configMap:
          items:
          - key: ca.crt
            path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
          - fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
            path: namespace
      - configMap:
          items:
          - key: service-ca.crt
            path: service-ca.crt
          name: openshift-service-ca.crt
status:
  .
  .
  .
```

so I guess this means we leave out setting runAsUser to 0 and `runAsRoot: true`, and we didn't have `allowPrivilegeEscalation` either so I'm guessing we don't need that either.

Reply:
so it lands in the "Anyuid" SCC which makes sense in that it's necessary so the container can run as UID 0, though i'm not entirely clear on *why* that SCC is selected given the securityContext the pod defines.

but i'll say this, it will quickly become apparent when you go to test this, whether or not you are defining the pod correctly such that a legacy catalog image can run successfully on a 4.12 cluster (in a PSA-privileged NS). 

I do still suspect you're going to have to set runAsUser: 0, in the end, to make it work.  Otherwise the pod will get assigned to the restricted-v2 SCC and get assigned a random uid, i think.  But do the experiment, if nothing else you'll learn some things.

as for whether you need to set "runAsNonRoot=false" on the pod (that is the official name of the securityContext for the field, for the record, not "runAsRoot"), you may not need to, but you probably should, if indeed the container is going to run as root.  No reason not to be explicit.

Reply:
@bparees,

> I do still suspect you're going to have to set runAsUser: 0, in the end, to make it work. Otherwise the pod will get assigned to the restricted-v2 SCC and get assigned a random uid, i think. But do the experiment, if nothing else you'll learn some things.

If we set AllowPrivilegeEscalation = true then is all sorted out
The restricted-v2 SCC policy cannot be applied and we granted the right permissions
I do not think that we can set RunAsUser = 0 at all. 
Therefore, it will work as before ( we will have a anyuid that has access to the root filesystem )

c/c @anik120

Reply:
I brought up a 4.10 cluster and see that the default marketplace catalogs are running as uid 1001, not 0 (as @camilamacedo86 alluded to).  So among other things, i think that means "runAsRoot" is the wrong name for the config field.  And yes, it means we should not be setting `runAsUser: 0` in the pod.  (I don't think it would fail, since the catalog controller has permission to use the "anyuid" SCC, so it can probably create pods that run as uid 0, but it doesn't seem like that's what we want it to do here).

Reply:
@camilamacedo86 @bparees updated the PR, not running as 0 anymore, renamed the field to `requirePrivilegedAcess`, and only setting AllowPrivilegeEscalation = true and readOnlyRootFilesystem: false when `requirePrivilegedAccess` is set to true. PTAL.